src/stactools/landsat/stac.py

```diff
@@ -63,7 +64,16 @@ def create_item(
     """
     base_href = "_".join(mtl_xml_href.split("_")[:-1])
 
-    mtl_metadata = MtlMetadata.from_file(mtl_xml_href, read_href_modifier)
+    mtl_metadata = None
+
+    if urlparse(mtl_xml_href).path.endswith(".txt"):
+        try:
+            mtl_metadata = MtlMetadata.from_text_file(mtl_xml_href, read_href_modifier)
+        except Exception as e:
+            logger.warning(f"Failed to parse MTL from text: {e}")
+
+    if mtl_metadata is None:
+        mtl_metadata = MtlMetadata.from_file(mtl_xml_href, read_href_modifier)
```

Comment:
This feels a little funny to me, since the argument name is `mtl_xml_href`. I'm also 👎🏼  on changing behavior based on filename extension, as that can be fragile. I'd prefer a new function, e.g.:

```python
def create_item_from_mtl_text(mtl_text_href: str, ...):
    mtl_metadata = MtlMetadata.from_text_file(mtl_text_href, read_href_modifier)
    return create_item_from_mtl_metadata(mtl_metadata)  # could be a new `MtlMetadata.create_item` method, dealer's choice
```